generate_toc.sh

```diff
@@ -0,0 +1,46 @@
+#!/bin/bash
```

Comment:
This script has 6 [SC2086](https://github.com/koalaman/shellcheck/wiki/SC2086) and 1 [SC2048](https://github.com/koalaman/shellcheck/wiki/SC2048) in a shellcheck run, but none of them are critical blockers (since anyone with whitespace in their home directory name will be essentially unable to use any shell script at Mozilla anyways!).